requirements.txt

```diff
@@ -1,23 +0,0 @@
-# These are known good versions. You're free to use others and things will
-# likely work, but no promises are made, especilly if you go older.
-PyYaml==3.12
-azure-mgmt-dns==1.0.1
-azure-common==1.1.6
-boto3==1.4.6
-botocore==1.6.8
-dnspython==1.15.0
-docutils==0.14
-dyn==1.8.0
-futures==3.1.1
-google-cloud==0.27.0
-incf.countryutils==1.0
-ipaddress==1.0.18
-jmespath==0.9.3
-msrestazure==0.4.10
-natsort==5.0.3
-nsone==0.9.14
-ovh==0.4.7
-python-dateutil==2.6.1
-requests==2.13.0
-s3transfer==0.1.10
-six==1.10.0
```

Comment:
This is the full/complete list of requirements for octoDNS if you used 100% of it's providers, not necessarily the ones that someone would install when running it and only using a subset. There was a separate minimal list in setup.py for this reason and the expectation was that users would explicitly install the subset of these needed for their providers. If all of this moves to a single list in setup.cfg that'd force everyone to install everything all the time. I don't want to do that since some of the providers, Google Cloud specifically, install dozens of indirect requirements. 

Not sure, but maybe that means there needs to be 3 lists in setup.cfg or maybe just 2? One with the base set that's always installed as hard-requirements, and then the dev set would get the 100% coverage provider requirements in addition to the dev tools stuff.

Reply:
Also makes me realized that the providers should probably either list out their requirements or check for them and complain if they're not installed. Longer term I may break up the more involved providers into their own modules, but don't have specific plans/timeline for that yet.

Reply:
I agree your remarks, read me comment and my propositions below.